src/transaction.ts

```diff
@@ -562,23 +596,27 @@ class Transaction extends DatastoreRequest {
     if (options.transactionOptions) {
       reqOpts.transactionOptions = options.transactionOptions;
     }
-
-    this.request_(
-      {
-        client: 'DatastoreClient',
-        method: 'beginTransaction',
-        reqOpts,
-        gaxOpts: options.gaxOptions,
-      },
-      (err, resp) => {
-        if (err) {
-          callback(err, null, resp);
-          return;
+    const promiseFunction: RequestAsPromiseCallback = (
+      resolve: RequestResolveFunction
+    ) => {
+      this.request_(
+        {
+          client: 'DatastoreClient',
+          method: 'beginTransaction',
+          reqOpts,
+          gaxOpts: options.gaxOptions,
+        },
+        // In original functionality sometimes a response is provided when an error is also provided
+        // reject only allows us to pass back an error so use resolve for both error and non-error cases.
```

Comment:
What do you mean by original functionality here? Is this comment describing how it worked before the change?

Reply:
Yeah, I'd say don't reference the way the code was before, make sure the comments explain the current state of the program

Reply:
Sounds good. Comment has been changed.